File: Tyche.iOS/Pool/Sources/Pool/Infrastructure/PoolGamblerScoreRemoteRepository.swift
import Core

class PoolGamblerScoreRemoteRepository : PoolGamblerScoreRepository {
    private let poolGamblerScoreRemoteDataSource: PoolGamblerScoreRemoteDataSource
    
    private let networkErrorHandler: NetworkErrorHandler
    
    init(
        poolGamblerScoreRemoteDataSource: PoolGamblerScoreRemoteDataSource,
        networkErrorHandler: NetworkErrorHandler
    ) {
        self.poolGamblerScoreRemoteDataSource = poolGamblerScoreRemoteDataSource
        self.networkErrorHandler = networkErrorHandler
    }
    
    func getPoolGamblerScoresByGambler(
        gamblerId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerScore>, Error> {
        return await networkErrorHandler.handle {
            let page = try await poolGamblerScoreRemoteDataSource.getPoolGamblerScoresByGambler(
                gamblerId: gamblerId,
                next: next,
                searchText: searchText
            )
            return page.map { poolGamblerScoreResponse in
                poolGamblerScoreResponse.toPoolGamblerScore()
            }
        }
    }
    
    func getPoolGamblerScoresByPool(
        poolId: String,
        next: String?,
        searchText: String?
    ) async -> Result<CursorPage<PoolGamblerScore>, Error> {
        return await networkErrorHandler.handle {
            let page = try await poolGamblerScoreRemoteDataSource.getPoolGamblerScoresByPool(
                poolId: poolId,
                next: next,
                searchText: searchText
            )
            return page.map { poolGamblerScoreResponse in
                poolGamblerScoreResponse.toPoolGamblerScore()
            }
        }
    }
}

File: Tyche.Android/data/pool/src/main/java/com/felipearpa/data/pool/infrastructure/PoolGamblerScoreRemoteRepository.kt
package com.felipearpa.data.pool.infrastructure

import com.felipearpa.data.pool.domain.PoolGamblerScore
import com.felipearpa.data.pool.domain.PoolGamblerScoreRemoteDataSource
import com.felipearpa.data.pool.domain.PoolGamblerScoreRepository
import com.felipearpa.data.pool.domain.toPoolGamblerScore
import com.felipearpa.tyche.core.network.NetworkExceptionHandler
import com.felipearpa.tyche.core.paging.CursorPage
import javax.inject.Inject

internal class PoolGamblerScoreRemoteRepository @Inject constructor(
    private val poolGamblerScoreRemoteDataSource: PoolGamblerScoreRemoteDataSource,
    private val networkExceptionHandler: NetworkExceptionHandler
) :
    PoolGamblerScoreRepository {

    override suspend fun getPoolGamblerScoresByGambler(
        gamblerId: String,
        next: String?,
        searchText: String?
    ): Result<CursorPage<PoolGamblerScore>> {
        return networkExceptionHandler.handle {
            poolGamblerScoreRemoteDataSource.getPoolGamblerScoresByGambler(
                gamblerId = gamblerId,
                next = next,
                searchText = searchText
            ).map { gamblerPoolResponse -> gamblerPoolResponse.toPoolGamblerScore() }
        }
    }

    override suspend fun getPoolGamblerScoresByPool(
        poolId: String,
        next: String?,
        searchText: String?
    ): Result<CursorPage<PoolGamblerScore>> {
        return networkExceptionHandler.handle {
            poolGamblerScoreRemoteDataSource.getPoolGamblerScoresByPool(
                poolId = poolId,
                next = next,
                searchText = searchText
            ).map { gamblerPoolResponse -> gamblerPoolResponse.toPoolGamblerScore() }
        }
    }

    override suspend fun getPoolGamblerScore(
        poolId: String,
        gamblerId: String
    ): Result<PoolGamblerScore> {
        return networkExceptionHandler.handle {
            poolGamblerScoreRemoteDataSource.getPoolGamblerScore(
                poolId = poolId,
                gamblerId = gamblerId
            ).toPoolGamblerScore()
        }
    }
}